src/relay/op/tensor/transform.cc

```diff
@@ -498,5 +498,66 @@ and type as the input array.
 .set_support_level(3)
 .add_type_rel("FullLike", FullLikeRel);
 
+// Have no idea how to assert the constraint.
+// CollapseSumLike: <A, B> -> B where BroadCast(A, B) = A
+bool CollapseSumLikeRel(const Array<Type>& types,
+                        int num_inputs,
+                        const Attrs& attrs,
+                        const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 3);
+  reporter->Assign(types[2], types[1]);
```

Comment:
Maybe see ReduceShapeImpl in https://github.com/dmlc/tvm/blob/master/nnvm/src/top/tensor/reduce.cc could provide guidance on how the axis params are supposed to work? (Strangely, `collapse_sum` itself doesn't use the reduction.)

Reply:
I am confused as well - I originally thought I need reduce_shape (which was at another pr), so at the mean time I can get this up.